src/posts/lucky_0.28.cr

```diff
@@ -0,0 +1,326 @@
+    ```crystal
+    table do
+      column data : ReportData, serialize: true
+    end
+
+    class ReportData
+      include JSON::Serializable
+      property key : String
+    end
+    ```
+
+    Our jsonb columns now have more type-safety!
+
+    [read more](https://github.com/luckyframework/avram/pull/695) in the PR.
+
+    ### Support for subdomains
+
+    In your application, you may have sections of your site that require a subdomain
+    to access. For example, a separate API with `api.yoursite.com`, or maybe logging
+    in to the site sends you to a `dashboard.yoursite.com`.
+
+    Before this update, you had to use `request.hostname.as(String).split('.').first`
+    to get the subdomain value, then use a before pipe to direct your traffic to the
+    correct location. Now we have built-in subdomain support to handle this.
+
+    ```crystal
+    class Users::Index < BrowserAction
+      # yoursite.com/users
+      get "/users" do
+        #...
+      end
+    end
+
+    class ApiAction < Lucky::Action
+      # Include this module in your actions
+      # that need subdomains
+      include Lucky::Subdomain
+    end
+
+    class Api::Users::Index < ApiAction
+      require_subdomain "api"
+
+      # api.yoursite.com/api/users
+      get "/api/users" do
+        #...
+      end
+    end
+    ```
+
+    This can have so many uses, like only access these actions under the "qa" subdomain,
+    or a multitenant application, or even just a staging environment.
+
+    [read more](https://github.com/luckyframework/lucky/pull/1537) in the PR.
+
+    ### Action updates
+
+    Actions got a few small updates starting with a new `multipart?` helper
+    method that returns `true` when the request is multipart (i.e. uploading a file).
+    This method accompanies the already existing methods `ajax?`, `xml?`, `plain_text?` and
+    a few others.
+
+    Along with the new helper methods, there's also two new response methods, `raw_json`,
+    and `html_with_status`.
+
+    The current `json()` response method is used to take a serializable object, and convert
+    that to a JSON response. In some cases, for example GraphQL, you may have a raw JSON string.
+    You couldn't use `json()` because this would convert your string in to an escaped json string.
+    Now you don't have to build out the response manually. Just use `raw_json()` instead.
+
+    Lastly is the `html_with_status`. More often than not, if you're rending an HTML page, it's
+    going to be a 200 response from the server. Lucky handles the error responses like 404, 422, and
+    500 for you in a separate action, so there's usually no reason to change this.
+
+    However, Lucky believes in having an escape hatch for those times you need to do non-standard things.
+    If you need to render an HTML page that's not a 200 response, you can use `html_with_status`
+
+    ```crystal
+    get "/tea_party" do
+      # render the IndexPage with status 418
+      html_with_status IndexPage, 418
```

Comment:
Do we have nice helper methods like `Lucky::Status::ImATeapot`? That might be nicer than a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)).

Reply:
This method requires a magic number 😅  I guess I could use the enum and call `.value` on it 🤔